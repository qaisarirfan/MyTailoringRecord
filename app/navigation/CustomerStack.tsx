import React from "react";
import { createNativeStackNavigator } from "@react-navigation/native-stack";
import CustomerDetail from "../screens/secure/customer-detail";
import AddMeasurement from "../screens/secure/add-measurement";
import AdditionalCustomizationOptions from "../screens/secure/additional-customization-options";
import { DeliveryScheduling } from "../screens/secure/delivery-scheduling";
import { RouteProp, useRoute } from "@react-navigation/native";
import { CustomerStackParamList } from "../hooks/useNavigation";

const Stack = createNativeStackNavigator<CustomerStackParamList>();

type Route = RouteProp<CustomerStackParamList, "CustomerDetail">;

export default function CustomerStack() {
  const { params } = useRoute<Route>();
  const { customerId } = params;

  return (
    <Stack.Navigator>
      <Stack.Screen
        name="CustomerDetail"
        component={CustomerDetail}
        options={{ title: "Customer" }}
        initialParams={{ customerId }}
      />
      <Stack.Screen
        name="AddMeasurement"
        component={AddMeasurement}
        options={{ title: "Measurement" }}
        initialParams={{ customerId }}
      />
      <Stack.Screen
        name="AdditionalCustomizationOptions"
        component={AdditionalCustomizationOptions}
        options={{ title: "Customization" }}
        initialParams={{ customerId }}
      />
      <Stack.Screen
        name="DeliveryScheduling"
        component={DeliveryScheduling}
        options={{ title: "Delivery Scheduling" }}
        initialParams={{ customerId }}
      />
    </Stack.Navigator>
  );
}
