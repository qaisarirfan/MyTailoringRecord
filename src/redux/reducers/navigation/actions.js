import { REDUCERS_NAME } from '../../../utils/constants';
import { createActionName } from '../../../utils/utility';

const { navigation } = REDUCERS_NAME;

export const SET_NAVIGATION = createActionName(navigation, 'SET_NAVIGATION');
