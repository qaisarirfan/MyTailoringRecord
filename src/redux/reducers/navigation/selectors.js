import get from 'lodash/get';
import {REDUCERS_NAME} from '../../../utils/constants';

const {navigation} = REDUCERS_NAME;

export const selectNavigation = state => get(state, `${navigation}.navigation`);
