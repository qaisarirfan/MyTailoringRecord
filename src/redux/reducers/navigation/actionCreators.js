import { SET_NAVIGATION } from './actions';

export function createNavigationAction(payload) {
  return {
    type: SET_NAVIGATION,
    payload,
  };
}
