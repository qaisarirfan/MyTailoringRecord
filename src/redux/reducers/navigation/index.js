import { createReducer } from '../../../utils/utility';
import { SET_NAVIGATION } from './actions';

// navigation initial state
export const initialState = {
  navigation: null,
};

// navigation Reducer
const reducers = {
  [SET_NAVIGATION](state, payload) {
    return {
      ...state,
      navigation: payload,
    };
  },
};

export default createReducer(reducers, initialState);
