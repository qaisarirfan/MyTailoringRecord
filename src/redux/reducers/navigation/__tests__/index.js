import navigation, {initialState} from ".."

describe("navigation reducer", () => {
  it("should return the initial state", () => {
    expect(navigation(undefined, {})).toEqual(initialState)
    expect(navigation(undefined, {})).toMatchSnapshot()
  })
  it("should handle GET_POST_SUCCESS", () => {})
  it("should handle UPDATE_POST_SUCCESS", () => {})
  it("should handle GET_POST_FAIL", () => {})
  it("should handle GET_POST_START", () => {})
})
