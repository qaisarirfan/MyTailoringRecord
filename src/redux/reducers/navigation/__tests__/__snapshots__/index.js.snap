// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navigation reducer should return the initial state 1`] = `
Object {
  "navigation": null,
}
`;
